### PR TITLE
LPS-165792 | master

### PR DESCRIPTION
**Pull request by sontruongces**
Hi team,

This is the fix for the [LPS-165792](https://issues.liferay.com/browse/LPS-165792), the original has passed my review at sontruongces#68. Please help me to review and give feedback.

Thanks!
cc: @TaiPhamLR

### CODE REVIEW

**Review by p-albuquerque — CHANGES_REQUESTED** [review]
This fix works only for 1-to-many situation. For many-to-many, the flow pass through `addObjectRelationshipMappingTableValues(...)` of **ObjectRelationshipLocalServiceImpl**, instead.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -3173,6 +3173,34 @@ private void _validateGroupId(long groupId, String scope)
...
+			SystemObjectDefinitionMetadata systemObjectDefinitionMetadata =
+				_systemObjectDefinitionMetadataRegistry.
+					getSystemObjectDefinitionMetadata(
+						objectDefinition.getName());
+
+			String externalReferenceCode =
+				systemObjectDefinitionMetadata.getExternalReferenceCode(
+					primaryKey);
+
+			if (externalReferenceCode == null) {
```
[the remark this answers is not in the document]
Comment (TaiPhamLR) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java]: Thanks for your review @p-albuquerque. I fixed them today, and also covered many-to-many case.
Please give your next feedback.

**Comment by naraandrade** [comment]
ci:test:sf

**Comment by naraandrade** [comment]
ci:test:object

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Just started reviewing :)

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi, 
I refactored the code a bit and resent it here: https://github.com/liferay-objects/liferay-portal/pull/1597

Let me know if you have questions or suggestions!

**Pull request closed without merging**